env0/utils.go

```diff
@@ -221,6 +221,94 @@ func writeResourceData(i interface{}, d *schema.ResourceData) error {
 	return nil
 }
 
+func getInterfaceValues(i interface{}) []interface{} {
+	var result []interface{}
+
+	val := reflect.ValueOf(i)
+
+	for i := 0; i < val.Len(); i++ {
+		field := val.Index(i)
+		result = append(result, field.Interface())
+	}
+
+	return result
+}
+
+// Extracts values from a slice of interfaces, and writes it to resourcedata at name.
+func writeResourceDataSlice(i interface{}, name string, d *schema.ResourceData) error {
+	writeResourceDataSliceStruct := func(val reflect.Value, value map[string]interface{}) {
+		for i := 0; i < val.NumField(); i++ {
+			fieldName := val.Type().Field(i).Name
+			fieldNameSC := toSnakeCase(fieldName)
+			field := val.Field(i)
+			fieldType := field.Type()
+
+			if fieldType.Kind() == reflect.Ptr {
+				if field.IsNil() {
+					continue
+				}
+
+				field = field.Elem()
+			}
+
+			if d.Get(name+".*."+fieldNameSC) == nil {
+				continue
+			}
+
+			value[fieldNameSC] = field.Interface()
+		}
+	}
+
+	ivalues := getInterfaceValues(i)
+
+	var values []interface{}
+
+	for _, ivalue := range ivalues {
+		value := make(map[string]interface{})
+
+		val := reflect.ValueOf(ivalue)
+
+		for i := 0; i < val.NumField(); i++ {
+			fieldName := val.Type().Field(i).Name
+
+			// Assumes golang is CamalCase and Terraform is snake_case.
+
+			fieldNameSC := toSnakeCase(fieldName)
+
+			field := val.Field(i)
+
+			fieldType := field.Type()
+
+			if fieldType.Kind() == reflect.Ptr {
+				if field.IsNil() {
+					continue
+				}
+
+				field = field.Elem()
+				fieldType = field.Type()
+			}
+
+			if fieldType.Kind() == reflect.Struct {
+				writeResourceDataSliceStruct(field, value)
+			}
+
+			if d.Get(name+".*."+fieldNameSC) == nil {
```

Comment:
🟢
I guess that you're checking here that the property exist in the destination schema, maybe add a short comment above that?.

And why don't you check that at first?

and if the fieldName is a "direct child", what's the purpose of `.*` between the name and the fieldName?

Reply:
this has been moved to a different PR.